### ISSUE TITLE
Instances Query API fails with SourceSelector with empty properties list

### ISSUE THREAD

**Issue by andoppomny** [cognitedata/cognite-sdk-python#1660]
In the Instances Query API, using `Select([SourceSelector(view_id, [])])` no longer works; properties must be non-empty list. It produces the following:

> Request had 1 constraint violations. Please fix the request and try again. [properties must not be null] | code: 400 | X-Request-ID: 8daf71e9-f36f-959b-9a93-c5c48415957f

This used to work, and is a bug since the user might not be interested in the properties. A node might not even have any properties which would make it unselectable.

The code
```
class SourceSelector(CogniteObject):
    source: ViewId
    properties: list[str] | None = None
    target_units: list[TargetUnit] | None = None

    def dump(self, camel_case: bool = True) -> dict[str, Any]:
        output: dict[str, Any] = {"source": self.source.dump(camel_case)}
        if self.properties:
            output["properties"] = self.properties
```
from https://github.com/cognitedata/cognite-sdk-python/blob/17134436a9712e439a1c5f2314594d6733f61ade/cognite/client/data_classes/data_modeling/query.py#L37C14-L37C20

is the offender, as it omits the "properties" field from the http request when the properties list is falsy (empty).
The server however requires this field to be present.

Also, the error message is not very helpful, as it doesn't mention which part of the query the constraint is violated for.

**Comment by haakonvt** [comment]
Hi @andoppomny and thanks for raising the issue! It seems to be a regression from PR #1642 when unit support was added. I've made a PR fix here: https://github.com/cognitedata/cognite-sdk-python/pull/1661 and I'll notify you once a new release has been made.

**Comment by haakonvt** [comment]
@andoppomny released now as part of `7.26.2`, let me know if everything works as expected

**Comment by haakonvt** [comment]
Please reopen if the issue still persist!
